code/modules/mob/living/living.dm

```diff
@@ -57,6 +57,9 @@
 	if(moving_diagonally) //no mob swap during diagonal moves.
 		return 1
 
+	if(ispet(src))
```

Comment:
Don't use `istype(src)` checks in procs. Override this on pets. If that's infeasible, you need to split this proc into new procs and then override this proc on pets.